app/scripts/metamask-controller.js

```diff
@@ -571,7 +565,7 @@ export default class MetamaskController extends EventEmitter {
       onPreferencesStateChange: this.preferencesController.store.subscribe.bind(
         this.preferencesController.store,
       ),
-      onNetworkStateChange: networkControllerMessenger.subscribe.bind(
+      onNetworkDidChange: networkControllerMessenger.subscribe.bind(
         networkControllerMessenger,
         'NetworkController:stateChange',
       ),
```

Comment:
A few lines below here, the `getERC20TokenName` option should be removed. This option is not accepted as of v21

>BREAKING: TokensController.watchAsset now performs on-chain validation of the asset's symbol and decimals, if they're defined in the contract (https://github.com/MetaMask/core/pull/1745)
>
>    The TokensController constructor no longer accepts a getERC20TokenName option. It was no longer needed due to this change.

Reply:
Fixed here https://github.com/MetaMask/metamask-extension/pull/22372